pandas/core/arrays/boolean.py

```diff
@@ -286,6 +286,19 @@ def _from_sequence(cls, scalars, dtype=None, copy: bool = False):
         values, mask = coerce_to_array(scalars, copy=copy)
         return BooleanArray(values, mask)
 
+    @classmethod
+    def _from_sequence_of_strings(cls, strings, dtype=None, copy=False):
+        def map_string(s):
+            if s in ["True", "true", "1"]:
+                return True
+            elif s in ["False", "false", "0"]:
+                return False
+            else:
+                return s
```

Comment:
At some point, we should probably make a more performant implementation for this.

Reply:
Agreed, any thoughts on what such an implementation would look like?

Reply:
IMO, perf isn't worth worrying about too much until we can push the actual parsing down to the C reader. As is, we've already built up a list of Python strings in memory.

Reply:
this is actually not too hard to do in parsers.pyx; this is generically handled in python code rather than in cython.